### PR TITLE
feat(ios): pass clearTransactionIos function to context

### PR DESCRIPTION
**Pull request by mmajk55**
Pass `clearTransactionIos` to Iap context

### CODE REVIEW

File: src/hooks/withIAPContext.tsx [hunk 1]
```diff
@@ -6,6 +6,7 @@ import {
   promotedProductListener,
   purchaseErrorListener,
   purchaseUpdatedListener,
+  clearTransactionIOS,
```
Comment (jeremybarbet) [inline src/hooks/withIAPContext.tsx:9]: You don't need to edit the withIapContext file, the function can be imported straight away in the useIAP hook similar as `finishTransaction` and `finishTransaction as iapFinishTransaction`

File: src/hooks/useIAP.ts [hunk 2]
```diff
@@ -34,6 +34,7 @@ type IAP_STATUS = {
   getSubscriptions: (skus: string[]) => Promise<void>;
   requestPurchase: typeof iapRequestPurchase;
   requestSubscription: typeof iapRequestSubscription;
+  clearTransactionIOS: () => Promise<void>
```
Comment (jeremybarbet) [inline src/hooks/useIAP.ts:37]: Can you move this under the `finishTransaction` method to group by "logic" methods, and you can import it straight away from this file as 

```ts
import {clearTransactionIOS as iapClearTransactionIOS} from './iap'
```

**Review by jeremybarbet — CHANGES_REQUESTED** [review]

**Comment by jeremybarbet** [comment]
A few comments, also, once done, make sure to run `yarn format` to apply prettier to your file and/or `yarn lint` to run tsc/prettier/eslint to your changes

File: src/hooks/withIAPContext.tsx (hunk 1 above)
Comment (mmajk55) [reply src/hooks/withIAPContext.tsx:9]: true 👍

**Review by hyochan — APPROVED** [review]
LGTM!

However, I want to point out that having this function exposed in the hook is a bit redundant since this can be called using `clearTransactionIOS` function in `react-native-iap` itself.

The functions exposed in the hook are intended as they needed the hook's lifecycle, such as having products updated declaratively when fetching them.

**Comment by jeremybarbet** [comment]
@hyochan I agree on that, I was gonna make this comment as well. Not sure if it's worth it exposing through the hook as well

**Comment by jeremybarbet** [comment]
@hyochan We already expose `requestPurchase` and `requestSubscription` which are not doing anything through the context. To keep consistency I would probably remove everything from the hook that doesn't have any interaction with the context

**Comment by hyochan** [comment]
> @hyochan We already expose `requestPurchase` and `requestSubscription` which are not doing anything through the context. To keep consistency I would probably remove everything from the hook that doesn't have any interaction with the context

Thanks for clarifying this. I think what's most important is the documentation since there will be more contributors coming to update similar changes. I can work on the documentation later.

**Comment by mmajk55** [comment]
I didn't know that this function can be called directly from RIap. Thanks for clarifying. So I think that we can close this PR

**Comment by jeremybarbet** [comment]
> > @hyochan We already expose `requestPurchase` and `requestSubscription` which are not doing anything through the context. To keep consistency I would probably remove everything from the hook that doesn't have any interaction with the context
> 
> Thanks for clarifying this. I think what's most important is the documentation since there will be more contributors coming to update similar changes. I can work on the documentation later.

I started reworking the documentation on this [PR](https://github.com/dooboolab/react-native-iap/pull/1815) Feel free to jump in and add to it! (I will revert my "wip" commit, so it has a better commit message)

**Comment by jeremybarbet** [comment]
[I removed ](https://github.com/dooboolab/react-native-iap/pull/1797/commits/770858645a4fd465a63e5b057b0e1a9d56bf321f)`requestPurchase` and `requestSubscription` from the hook to keep consistency.

**Pull request closed without merging**